core/src/main/scala/org/apache/spark/api/r/RRDD.scala

```diff
@@ -127,6 +127,13 @@ private[r] object RRDD {
       sparkConf.setExecutorEnv(name.toString, value.toString)
     }
 
+    if (sparkEnvirMap.containsKey("spark.r.sql.default.derby.dir") &&
```

Comment:
setting `derby.stream.error.file` is all we need to move `derby.log` - I'd like to proceed with this change to make the cut for 2.1.1 release unless we have serious concern?